crates/ruff_benchmark/benches/ty_walltime.rs

```diff
@@ -0,0 +1,204 @@
+use divan::{Bencher, bench_group};
+
+use rayon::ThreadPoolBuilder;
+use ruff_benchmark::real_world_projects::RealWorldProject;
+use ruff_db::system::{OsSystem, System, SystemPath, SystemPathBuf};
+
+use ty_project::metadata::options::{EnvironmentOptions, Options};
+use ty_project::metadata::value::{RangedValue, RelativePathBuf};
+use ty_project::{Db, ProjectDatabase, ProjectMetadata};
+
+static RAYON_INITIALIZED: std::sync::Once = std::sync::Once::new();
+
+fn setup_rayon() {
+    // Initialize the rayon thread pool outside the benchmark because it has a significant cost.
+    // Ideally, we wouldn't have to do this but there's a significant variance
+    // if we run the benchmarks multi threaded:
+    // ```
+    // ty_walltime        fastest       │ slowest       │ median        │ mean          │ samples │ iters
+    // ╰─ colour_science 153.7 ms       │ 2.177 s       │ 2.106 s       │ 1.921 s       │ 10      │ 10
```

Comment:
unclosed code fence :-)

````suggestion
    // if we run the benchmarks multi-threaded:
    // ```
    // ty_walltime        fastest       │ slowest       │ median        │ mean          │ samples │ iters
    // ╰─ colour_science 153.7 ms       │ 2.177 s       │ 2.106 s       │ 1.921 s       │ 10      │ 10
    // ```
````